test/distributed/test_c10d_nccl.py

```diff
@@ -235,45 +252,61 @@ def test_empty_tensors(self):
         pg.reduce(xs).wait()
         self.assertEqual(0, xs[0].numel())
 
-        ys = [[torch.cuda.FloatTensor([]) for _ in range(self.world_size)]]
+        ys = [[torch.FloatTensor([]).cuda(local_device_idx) for _ in range(self.world_size)]]
         pg.allgather(ys, xs).wait()
         for y in ys[0]:
             self.assertEqual(0, y.numel())
 
-        ys = [torch.cuda.FloatTensor([])]
-        xs = [[torch.cuda.FloatTensor([]) for _ in range(self.world_size)]]
+        ys = [torch.FloatTensor([]).cuda(local_device_idx)]
+        xs = [[torch.FloatTensor([]).cuda(local_device_idx) for _ in range(self.world_size)]]
         pg.reduce_scatter(ys, xs).wait()
         self.assertEqual(0, ys[0].numel())
 
     @requires_nccl()
     @sandcastle_skip_if(torch.cuda.device_count() < 2, "NCCL test requires 2+ GPUs")
     def test_broadcast_ops(self):
-        store = c10d.FileStore(self.file.name, self.world_size)
-        pg = c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
+        store = c10d.FileStore(self.file_name, self.world_size)
+        pg = self._create_process_group_nccl(store, self.rank, self.world_size, self.opts())
+        rank_to_GPU = init_multigpu_helper(self.world_size, "nccl")
 
         def broadcast(xs, rootRank, rootTensor):
             opts = c10d.BroadcastOptions()
             opts.rootRank = rootRank
             opts.rootTensor = rootTensor
-            work = pg.broadcast(xs, opts)
-            work.wait()
+            fut = pg.broadcast(xs, opts).get_future()
+            fut.wait()
+            return fut.value()
+
+        # Every rank is root once
+        for i in range(self.world_size):
+            # Run with 1 input tensor
+            x = torch.tensor([self.rank]).cuda(rank_to_GPU[self.rank][0])
+            output = broadcast([x], i, 0)
+            # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
+            self.assertEqualIgnoreType(torch.tensor([i]), output[0])
 
-        # for every root tensor
-        for rt in range(self.num_gpus):
-            tensors = []
-            for i in range(self.num_gpus):
-                tensors.append(torch.tensor([i]).cuda(i))
+            expected_tensor = torch.empty([i + 1, i + 1]).fill_(i + 1)
+            xs = [torch.empty([i + 1, i + 1]).fill_(-1).cuda(device=device_idx) for device_idx in rank_to_GPU[self.rank]]
+
+            # test with multiple input tensors
```

Comment:
If understand this correctly, this means we have multiple GPUs in one rank, right? If so, can we make the comment a little bit clearer?